File: backend/schemas/photometry.py
from pydantic import BaseModel


class PhotometryRequest(BaseModel):
    target_id: str
    observation_ids: list[str]
    aperture_radius: float = 5.0
    inner_annulus: float = 10.0
    outer_annulus: float = 15.0


class PhotometryMeasurement(BaseModel):
    observation_id: str
    epoch: str
    hjd: float
    raw_flux: float
    sky_flux: float
    net_flux: float
    instrumental_mag: float
    mag_error: float


class PhotometryResponse(BaseModel):
    target_id: str
    aperture_radius: float
    measurements: list[PhotometryMeasurement]

File: frontend/src/types/photometry.ts
export interface PhotometryRequest {
  target_id: string;
  observation_ids: string[];
  aperture_radius: number;
  inner_annulus: number;
  outer_annulus: number;
}

export interface PhotometryMeasurement {
  observation_id: string;
  epoch: string;
  hjd: number;
  raw_flux: number;
  sky_flux: number;
  net_flux: number;
  instrumental_mag: number;
  mag_error: number;
}

export interface PhotometryResponse {
  target_id: string;
  aperture_radius: number;
  measurements: PhotometryMeasurement[];
}

export interface LightCurveRequest {
  target_id: string;
  observation_ids: string[];
  aperture_radius: number;
  inner_annulus: number;
  outer_annulus: number;
  fold_period: number | null;
}

export interface LightCurvePoint {
  hjd: number;
  phase: number | null;
  magnitude: number;
  mag_error: number;
}

export interface LightCurveResponse {
  target_id: string;
  period_days: number | null;
  points: LightCurvePoint[];
  x_label: string;
  y_label: string;
}
